src/sqlfluff/core/parser/lexer.py

```diff
@@ -560,6 +561,15 @@ def elements_to_segments(
                         )
                     )
 
+        # Add an end of file marker
+        segment_buffer.append(
```

Comment:
Related: Can you take a look at this code:
```
 ack --python is_final src
src/sqlfluff/core/rules/base.py
420:        Similar in spirit to BaseRule.is_final_segment(), but:
768:    def is_final_segment(self, context: RuleContext, filter_meta: bool = True) -> bool:

src/sqlfluff/rules/L052.py
275:        if not self.is_final_segment(context):
```

Seems like the function should be eliminated (or rewritten?) with the introduction of an `EndOfFile` segment.

Reply:
I can't see a reference to `is_final_segment` in L052, but I did find one use of the function on line 420 in `L003`. I've updated the rule to use `end_of_file` and got rid of `RuleContext.final_segment`.